### PR TITLE
dind: enable overlay storage

### PR DESCRIPTION
**Pull request by marun**
This change enables the overlay storage driver for the
docker-in-docker daemon if it is supported by the host.  overlay
should be faster than vfs by virtue of allowing copy-on-write.  vfs is
still the default if overlay is not available.

Thanks to @sttts and the kube PR that inspired this change: https://github.com/kubernetes/kubernetes/pull/27459/

cc: @openshift/networking

### CODE REVIEW

**Comment by marun** [comment]
[testextended][extended:networking]

**Comment by marun** [comment]
Flake #9444 - unable to retrieve image due to temporary resolution failure of gcr.io

re-[testextended][extended:networking]

**Comment by knobunc** [comment]
LGTM, thanks

**Comment by marun** [comment]
Minor update to remove the use of the dind wrapper script.

**Comment by danwinship** [comment]
I don't know much about overlayfs, but it seems to work for me (and keeps working if it I hack it so the overlayfs test will fail too)

**Comment by danwinship** [comment]
[merge]

File: contrib/vagrant/provision-util.sh [hunk 1]
```diff
@@ -399,3 +400,36 @@ os::provision::wait-for-node-config() {
...
     "${OS_WAIT_FOREVER}"
 }
+
+# Enable overlayfs for dind if it can be tested to work.
+os::provision::enable-overlay-storage() {
+  local storage_dir=${1:-/var/lib/docker}
+
+  local msg="WARNING: Unable to enable overlay storage for docker-in-docker"
+
+  if grep -q overlay /proc/filesystems; then
```
Comment (dcbw) [inline contrib/vagrant/provision-util.sh]: I don't have overlay in my /proc/filesystems on Fedora 23 by default.  Perhaps 'modprobe overlay || true' before checking if overlay exists?

Comment (dcbw) [inline contrib/vagrant/provision-util.sh]: > I don't have overlay in my /proc/filesystems on Fedora 23 by default. Perhaps 'modprobe overlay || true' before checking if overlay exists?

Nevermind on this one, it doesn't happen inside a vagrant VM with 'OPENSHIFT_DEV_DIND=1 vagrant up' but then again it's not supposed to.

**Comment by dcbw** [comment]
LGTM [merge]

**Pull request merged**